lib/scraper.js

```diff
@@ -127,9 +187,24 @@ Scraper.prototype.prepare = function prepare () {
 
 Scraper.prototype.load = function load () {
 	var self = this;
-	return Promise.map(self.originalResources, function loadResource (resource) {
-		return self.loadResource(resource);
+	_.forEach(self.originalResources, function loadResource (resource) {
+		self.loadResource(resource);
 	});
+	return self.waitForLoad();
+};
+
+Scraper.prototype.waitForLoad = function waitForLoad (previousCount) {
```

Comment:
Lets add comment about mechanism, because without comments it is pretty hard to understand what is going on